ui/camera.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+
+/** @typedef {import('./libapi.mjs').Context} Context */
+
+// To avoid errors for users who use SSR or Hybrid rendering (e.g. Nuxt.js), we need to check if we're in the browser.
+let initModule;
+
+async function initializeModule() {
+    if (typeof window !== 'undefined') {
+        const module = await import('./libapi.mjs');
+        initModule = module.default;
+        return initModule;
+    } else {
+        console.warn("web-gphoto2 is only available in the browser");
+        return null;
+    }
+}
+
+const ModulePromise = initializeModule().then(initModule => {
+    if (initModule) {
+        return initModule();
+    } else {
+        return null;
+    }
+});
+
+class Camera {
+    constructor() {
+        /** @type {Promise<unknown>} */
+        this.queue = Promise.resolve();
+        this.Module = null;
+        this.context = null;
+    }
+
+    rethrowIfCritical(err) {
+        // If it's precisely Error, it's a custom error; anything else - SyntaxError,
+        // WebAssembly.RuntimeError, TypeError, etc. - is treated as critical here.
+        if (err.constructor !== Error) {
+            throw err;
+        }
+    }
+
+    async connect() {
+        this.Module = await ModulePromise;
```

Comment:
Maybe lazily assign `ModulePromise` here? On one hand, this approach is faster, but on another if someone just imported the library but never connects to a camera, this can save some bandwidth / startup time.

Reply:
I'm not 100% sure if I implemented it correctly, but this is what I could come up with.